Assignment 2.Rmd

```diff
@@ -8,7 +8,9 @@ Now upload the data file "discipline-data.csv" as a data frame called "D1". Each
 Before you proceed, you will need to change the data type of the student id variable. Since it is a number R will automatically think it is an integer and code it as such (look at the list of variables by clicking on the data frame arrow in the Data pane. Here you will see the letters "int"" next to the stid variable, that stands for integer). However, in this case we are treating the variable as a category, there is no numeric meaning in the variable. So we need to change the format to be a category, what R calls a "factor". We can do this with the following code:
 
 ```{r}
+D1 <- read.csv("discipline-data.csv",header = TRUE)
 D1$stid <- as.factor(D1$stid)
+attach(D1)
```

Comment:
`attach()` is a bit of a tricky command, I would avoid it usually. It can save writing a bit of code but can cause a lot of confusion with respect to what dataframe you are referring to. The Google R Style Guide suggests to avoid to avoid it and I tend to agree.